<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="m600_pro">

  <xacro:include filename="$(find dji_m600_sim)/urdf/inertial.xacro" />
  <xacro:include filename="$(find dji_m600_sim)/urdf/joints.xacro" />
  <!-- <xacro:include filename="$(find dji_m600_sim)/urdf/transmissions.xacro" /> -->
  <xacro:include filename="$(find dji_m600_sim)/urdf/geometries.xacro" />
  <xacro:include filename="$(find dji_m600_sim)/urdf/properties.xacro" />
  <xacro:include filename="$(find dji_m600_sim)/urdf/gazebo.xacro" />


  <xacro:macro name="base_link">

   <link name="world"/>
   <link name="base_link"/>

    <joint name="vehicle_center_joint" type="fixed">
      <parent link="world"/>
      <child link="vehicle_center_link" />
      <origin xyz="${chassisXoffset} ${chassisYoffset} ${chassisZoffset}" rpy="0 0 0"/>
    </joint>


    <link name="vehicle_center_link">
      <visual>
        <geometry>
          <mesh filename="file://$(find dji_m600_sim)/meshes/m600.dae" scale="${chassisScale} ${chassisScale} ${chassisScale}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="${chassisMeshXoffset} ${chassisMeshYoffset} ${chassisMeshZoffset}" />
        <!-- <xacro:drone_cylinder /> -->
        <!-- <material name="blue" /> -->
      </visual>
      <collision>
        <xacro:box_geometry />
      </collision>
      <xacro:box_inertial/>
    </link>

    <joint name="vehicle_ground_joint" type="fixed">
      <parent link="vehicle_center_link"/>
      <child link="base_link" />
      <origin xyz="0 0 -${chassisZoffset}" rpy="0 0 0"/>
    </joint>


    <xacro:base_gazebo/>

  </xacro:macro>


    <!-- <xacro:macro name="imu">
    <link name="imu_link">
      <visual>
        <xacro:imu_geometry />
      </visual>
      <collision>
        <xacro:imu_geometry />
      </collision>
      <xacro:imu_inertial />
    </link>
    <xacro:imu_joint />

    <xacro:imu_gazebo />
  </xacro:macro> -->

</robot>
